### ISSUE TITLE
Generate error if positional-only parameter follows non-positional-only parameter using legacy syntax

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7290]
The [typing spec has been updated](https://github.com/python/typing/pull/1619/files) to clarify that type checkers should generate an error if a function signature uses the legacy mechanism to specify positional-only parameters and includes a positional-only parameter after a non-positional-only parameter it should be flagged as an error.

The follow should therefore result in a type error.

```python
def func(a: Any, __value: Any) -> Any:
    ...
```
